cluster-autoscaler/config/flags/flags.go

```diff
@@ -0,0 +1,525 @@
+	ignoreTaintsFlag                 = multiStringFlag("ignore-taint", "Specifies a taint to ignore in node templates when considering to scale a node group (Deprecated, use startup-taints instead)")
+	maxAutoprovisionedNodeGroupCount = flag.Int("max-autoprovisioned-node-group-count", 15, "The maximum number of autoprovisioned groups in the cluster.This flag is deprecated and will be removed in future releases.")
+	maxEmptyBulkDeleteFlag           = flag.Int("max-empty-bulk-delete", 10, "Maximum number of empty nodes that can be deleted at the same time. DEPRECATED: Use --max-scale-down-parallelism instead.")
+	nodeAutoprovisioningEnabled      = flag.Bool("node-autoprovisioning-enabled", false, "Should CA autoprovision node groups when needed.This flag is deprecated and will be removed in future releases.")
+)
+
+var autoscalingOptions *config.AutoscalingOptions
+
+// AutoscalingOptions returns the singleton instance of AutoscalingOptions, initializing it if necessary.
+func AutoscalingOptions() config.AutoscalingOptions {
+	if autoscalingOptions == nil {
+		newAutoscalingOptions := createAutoscalingOptions()
+		autoscalingOptions = &newAutoscalingOptions
+	}
+	return *autoscalingOptions
+}
+
+func createAutoscalingOptions() config.AutoscalingOptions {
+	minCoresTotal, maxCoresTotal, err := parseMinMaxFlag(*coresTotal)
+	if err != nil {
+		klog.Fatalf("Failed to parse flags: %v", err)
+	}
+	minMemoryTotal, maxMemoryTotal, err := parseMinMaxFlag(*memoryTotal)
+	if err != nil {
+		klog.Fatalf("Failed to parse flags: %v", err)
+	}
+	// Convert memory limits to bytes.
+	minMemoryTotal = minMemoryTotal * units.GiB
+	maxMemoryTotal = maxMemoryTotal * units.GiB
+
+	parsedGpuTotal, err := parseMultipleGpuLimits(*gpuTotal)
+	if err != nil {
+		klog.Fatalf("Failed to parse flags: %v", err)
+	}
+
+	// in order to avoid inconsistent deletion thresholds for the legacy planner and the new actuator, the max-empty-bulk-delete,
+	// and max-scale-down-parallelism flags must be set to the same value.
+	if isFlagPassed("max-empty-bulk-delete") && !isFlagPassed("max-scale-down-parallelism") {
+		*maxScaleDownParallelismFlag = *maxEmptyBulkDeleteFlag
+		klog.Warning("The max-empty-bulk-delete flag will be deprecated in k8s version 1.29. Please use max-scale-down-parallelism instead.")
+		klog.Infof("Setting max-scale-down-parallelism to %d, based on the max-empty-bulk-delete value %d", *maxScaleDownParallelismFlag, *maxEmptyBulkDeleteFlag)
+	} else if !isFlagPassed("max-empty-bulk-delete") && isFlagPassed("max-scale-down-parallelism") {
+		*maxEmptyBulkDeleteFlag = *maxScaleDownParallelismFlag
+	}
+
+	if isFlagPassed("node-autoprovisioning-enabled") {
+		klog.Warning("The node-autoprovisioning-enabled flag is deprecated and will be removed in k8s version 1.31.")
```

Comment:
We are approaching 1.33 release, perhaps these flags could be cleaned up?

Reply:
sure, what about the `UpdateNapEnabled` metric, should we remove it as well?

Reply:
Yup, I don't think it makes sense to keep a metric based on a deprecated flag.

Reply:
done